pkgs/tools/system/nvtop/build-nvtop.nix

```diff
@@ -0,0 +1,97 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, gtest
+, cudatoolkit
+, libdrm
+, ncurses
+, testers
+, udev
+, addOpenGLRunpath
+, amd ? false
+, intel ? false
+, msm ? false
+, nvidia ? false
+, apple ? false
+, panfrost ? false
+, panthor ? false
+, ascend ? false
+}:
+
+let
+  nvidia-postFixup = "addOpenGLRunpath $out/bin/nvtop";
+  libPath = lib.makeLibraryPath [ libdrm ncurses udev ];
+  drm-postFixup = ''
+    patchelf \
+      --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" \
+      --set-rpath "${libPath}" \
+      $out/bin/nvtop
+  '';
+  needDrm = (amd || msm || panfrost || panthor);
+  onOff = v: if v then "ON" else "OFF";
+in
+stdenv.mkDerivation (finalAttrs: {
+  pname = "nvtop";
+  version = "3.1.0";
+
+  src = fetchFromGitHub {
+    owner = "Syllo";
+    repo = "nvtop";
+    rev = finalAttrs.version;
+    hash = "sha256-MkkBY2PR6FZnmRMqv9MWqwPWRgixfkUQW5TWJtHEzwA=";
+  };
+
+  cmakeFlags = [
+    "-DBUILD_TESTING=ON"
+    "-DUSE_LIBUDEV_OVER_LIBSYSTEMD=ON"
+    "-DAMDGPU_SUPPORT=${onOff amd}"
+    "-DNVIDIA_SUPPORT=${onOff nvidia}"
+    "-DINTEL_SUPPORT=${onOff intel}"
+    "-DAPPLE_SUPPORT=${onOff apple}"
+    "-DMSM_SUPPORT=${onOff msm}"
+    "-DPANFROST_SUPPORT=${onOff panfrost}"
+    "-DPANTHOR_SUPPORT=${onOff panthor}"
+    "-DASCEND_SUPPORT=${onOff ascend}"
+  ] ++ lib.optionals nvidia [
+    "-DNVML_INCLUDE_DIRS=${cudatoolkit}/include"
+    "-DNVML_LIBRARIES=${cudatoolkit}/targets/x86_64-linux/lib/stubs/libnvidia-ml.so"
```

Comment:
i think this doesn't do anything after https://github.com/Syllo/nvtop/commit/8b6519eebc566b496e5d3a04ab0dc0c561f458dd

Reply:
Nice catch. I deleted this part and full package builds fine. Though I can't confirm that nvidia reporting actually work because I don't have nvidia GPU

Reply:
I can confirm nvidia support still works on nixos, but doesn't on nonnixos, not related to this PR